File: Problems/Algorithms/997. Find the Town Judge/FindTownJudge.kt
class Solution {
    fun findJudge(n: Int, trust: Array<IntArray>): Int {
        val len = trust.size-1
        if (n == 1) {
            if (trust.isEmpty()) return 1
            return -1
        }
        
        val mapIn = mutableMapOf<Int, Int>()
        val mapOut = mutableMapOf<Int, Int>()
        
        for (i in 0..len) {
            val a: Int = trust[i][0]
            val b: Int = trust[i][1]
            mapIn.put(b, mapIn.getOrDefault(b, 0) + 1)
            mapOut.put(a, mapOut.getOrDefault(a, 0) + 1)
        }
        
        for (key in mapIn.keys) {
            if (mapIn[key] == n-1 && !mapOut.containsKey(key)) return key 
        }
        
        return -1
    }
}

File: Problems/Algorithms/997. Find the Town Judge/find-town-judge.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        vector<int> in(n+1), out(n+1);
        for (auto relation : trust) {
            in[relation[1]]++;
            out[relation[0]]++;
        }

        for (int i = 1; i <= n; i++) {
            if (in[i] == n-1 && out[i] == 0) {
                return i;
            }
        }
        return -1;
    }
};
